lib/src/event_controller.dart

```diff
@@ -173,10 +177,14 @@ class EventController<T extends Object?> extends ChangeNotifier {
         });
       } else {
         _calendarData.events[date]!.add(event);
+        _calendarData.events[date]!.sort((a, b) =>
+            a.startTime!.getTotalMinutes - b.startTime!.getTotalMinutes);
       }
     }
 
     _calendarData.eventList.add(event);
+    _calendarData.eventList.sort(
```

Comment:
Can we improve the sorting mechanism? This is way too costly. Can we improve this?

Reply:
Sure, I have improved and updated the PR.